android/src/main/java/com/mr/flutter/plugin/filepicker/FilePickerDelegate.java

```diff
@@ -232,8 +240,9 @@ private void startFileExplorer() {
         } else {
             if (type.equals("image/*")) {
                 intent = new Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+                intentChooser = Intent.createChooser(intent, this.imagePickerTitle != null ? this.imagePickerTitle : "Select a photo");
             } else {
-                intent = new Intent(Intent.ACTION_GET_CONTENT);
+                intent = new Intent(Intent.ACTION_OPEN_DOCUMENT);
```

Comment:
It seems this change breaks some older Android versions. For me with this change the file picker doesn't work on Android 9, but reverting this change or setting the intent type depending on Android version (like [here](https://github.com/dmitrio95/flutter_file_picker/commit/e7dcbff0e2240226fce69c468a6f23dacfe1fc92)) fixes this. I am not sure though which versions should use which intent type and why.

Reply:
It seems wrapping all intents with `createChooser` works with `launchMode="singleInstance"` regardless of the version of Android even with the old `ACTION_GET_CONTENT` intent type: [example](https://github.com/dmitrio95/flutter_file_picker/commit/cbdbf52a703082ad58b7ccd3e51c440b3eb05ab5). It will probably work with other launch modes too, but according to the description [here](https://developer.android.com/reference/android/content/Intent#ACTION_CHOOSER) it may force the system to show an extra activity chooser dialog if there are several possible activities for choosing a file. So perhaps this wrapping should better be somehow limited only to `singleInstance` applications.